lib/nerve.rb

```diff
@@ -15,35 +15,94 @@ class Nerve
 
     include Logging
 
-    def initialize(opts={})
+    def initialize(config_manager)
       log.info 'nerve: starting up!'
+      @config_manager = config_manager
 
       # set global variable for exit signal
       $EXIT = false
 
+      @watchers = {}
+      @watcher_versions = {}
+
+      # Flag to indicate a config reload is required by the main loop
+      # This decoupling is required for gracefully reloading config on SIGHUP
+      # as one should do basically nothing in a signal handler
+      @config_to_load = true
+      # Will be populated by load_config! in the main loop
+      @instance_id = nil
+      @services = {}
+      @heartbeat_path = nil
+
+      Signal.trap("HUP") do
+        @config_to_load = true
+      end
+
+     log.debug 'nerve: completed init'
+    end
+
+    def load_config!
+      log.info 'nerve: loading config'
+      @config_to_load = false
+      @config_manager.reload!
+      config = @config_manager.config
+
       # required options
       log.debug 'nerve: checking for required inputs'
       %w{instance_id services}.each do |required|
-        raise ArgumentError, "you need to specify required argument #{required}" unless opts[required]
+        raise ArgumentError, "you need to specify required argument #{required}" unless config[required]
```

Comment:
Yea, I mention this above as a concern. I'm currently trying out an apache like "check config" action but I'm not sure if I like that more than just logging an error and using the old configuration...